akka-stream/src/main/scala/akka/stream/scaladsl/Queue.scala

```diff
@@ -32,6 +32,23 @@ trait SourceQueue[T] {
 }
 
 /**
+ * This trait adds completion support to [[SourceQueue]].
+ */
+trait SourceQueueWithComplete[T] extends SourceQueue[T] {
+  /**
+   * Complete the stream normally. Use `watchCompletion` to be notified of this
+   * operation’s success.
+   */
+  def complete(): Unit
+
+  /**
+   * Complete the stream with a failure. Use `watchCompletion` to be notified of this
+   * operation’s success.
+   */
+  def fail(ex: Throwable): Unit
```

Comment:
Okay, so I gather that we do agree: it might be fixable in principle, but it is too much effort because all that is gained is that erroneous user code is flagged as such earlier than it is now (assuming test coverage).